src/test/rgw/bucket_notification/test_bn.py

```diff
@@ -1818,99 +1758,82 @@ def test_ps_s3_lifecycle_on_master():
-    topic_conf_list = [{'Id': notification_name,
-                        'TopicArn': topic_arn,
-                        'Events': ['s3:ObjectLifecycle:Expiration:*']
-                       }]
-    s3_notification_conf = PSNotificationS3(conn, bucket_name, topic_conf_list)
-    response, status = s3_notification_conf.set_config()
-    assert_equal(status/100, 2)
 
-    # start and abandon a multpart upload
-    obj_prefix = 'ooo'
-    content = 'bar'
+@attr('http_test')
+def test_lifecycle_abort_mpu():
+    """ test that when a multipart upload is aborted by lifecycle policy, http endpoint """
 
-    key_name = obj_prefix + str(1)
-    thr = threading.Thread(target = start_and_abandon_multipart_upload, args=(bucket, key_name, content,))
-    thr.start()
-    thr.join()    
+    def rules_creator(yesterday, obj_prefix):
+        return [
+            {
+                'ID': 'abort1',
+                'Filter': {'Prefix': obj_prefix},
+                'Status': 'Enabled',
+                'AbortIncompleteMultipartUpload': {'DaysAfterInitiation': 1},
+            }
+        ]
 
-    # create lifecycle policy -- assume rgw_lc_debug_interval=10 is in effect
-    client = boto3.client('s3',
-            endpoint_url='http://'+conn.host+':'+str(conn.port),
-            aws_access_key_id=conn.aws_access_key_id,
-            aws_secret_access_key=conn.aws_secret_access_key)
-    response = client.put_bucket_lifecycle_configuration(Bucket=bucket_name, 
-            LifecycleConfiguration={'Rules': [
-                {
-                    'ID': 'abort1',
-                    'Filter': {'Prefix': obj_prefix},
-                    'Status': 'Enabled',
-                    'AbortIncompleteMultipartUpload': {'DaysAfterInitiation': 1},
-                }
-            ]
-        }
-    )
+    def create_thread(bucket, obj_prefix, i, content):
+        key_name = obj_prefix + str(i)
+        return threading.Thread(target = start_and_abandon_multipart_upload, args=(bucket, key_name, content,))
 
-    # start lifecycle processing
-    admin(['lc', 'process'], get_config_cluster())
-    print('wait for 20s for the lifecycle...')
-    time.sleep(20)
+    conn = connection()
+    lifecycle('http', conn, 1, ['s3:ObjectLifecycle:Expiration:*'], create_thread, rules_creator,
+              ['ObjectLifecycle:Expiration:AbortMultipartUpload'], True)
 
-    wait_for_queue_to_drain(topic_name)
-    events = http_server.get_and_reset_events()
-    for event in events:
-        assert_equal(event['Records'][0]['eventName'], 'ObjectLifecycle:Expiration:AbortMPU')
-        assert key_name in event['Records'][0]['s3']['object']['key']
 
-    # cleanup
-    topic_conf.del_config()
-    s3_notification_conf.del_config(notification=notification_name)
-    # delete the bucket
-    conn.delete_bucket(bucket_name)
-    http_server.close()
 
 def ps_s3_creation_triggers_on_master(external_endpoint_address=None, ca_location=None, verify_ssl='true'):
```

Comment:
we can probably make that endpoint agnostic as well

Reply:
i think we can remove the external endpoint logic.
we can say that rabbitmq must be running (similar to how we do that with kafka), and not run it from within the test

Reply:
and aslo make `test_http_post_object_upload` endpoint agnostic

Reply:
does the **ca_location** work for all endpoint types?

Reply:
not really. this test is actually mixing two unrelated things - testing triggers and testing amqp security.
i think you are right here. lets keep that out-of-scope for now